framework/render/core.py

```diff
@@ -19,3 +27,40 @@ def init_mfr(app):
         'ASSETS_FOLDER': os.path.join(app.static_folder, 'mfr'),
     })
     mfr.collect_static(dest=mfr.config['ASSETS_FOLDER'])
+
+
+def render_is_done_or_happening(cache_path, temp_path):
+    # if the cached file exists do nothing
+    if os.path.isfile(cache_path):
+        return True
+
+    if os.path.isfile(temp_path):
+        if time.time() - os.path.getmtime(temp_path) > settings.MFR_TIMEOUT:
+            # If the temp path has not been modified since the timeout
+            # seconds assume the task failed, remove the file and
+            # start over
+            os.remove(temp_path)
+            return False
+        # Otherwise the task is happening somewhere else
+        return True
+
+    # If none of the above go ahead and start
+    return False
+
+
+def save_to_file_or_error(download_url, dest_path):
+    with open(dest_path, 'wb') as temp_file:
+        response = requests.get(download_url, stream=True)
+        if response.ok:
+            for block in response.iter_content(1024):  # 1kb
+                if not block:
```

Comment:
This check isn't necessary--`iter_content` returns a generator that will exhaust when the response is empty.